github/actions_cache.go

```diff
@@ -0,0 +1,236 @@
+type ActionsCache struct {
+	ID             *int64     `json:"id,omitempty" url:"-"`
+	Ref            *string    `json:"ref,omitempty" url:"ref"`
+	Key            *string    `json:"key,omitempty" url:"key"`
+	Version        *string    `json:"version,omitempty" url:"-"`
+	LastAccessedAt *Timestamp `json:"last_accessed_at,omitempty" url:"-"`
+	CreatedAt      *Timestamp `json:"created_at,omitempty" url:"-"`
+	SizeInBytes    *int64     `json:"size_in_bytes,omitempty" url:"-"`
+}
+
+// ActionsCacheList represents a list of GitHub actions Cache.
+//
+// GitHub API docs: https://docs.github.com/en/rest/actions/cache?apiVersion=2022-11-28#list-github-actions-caches-for-a-repository
+type ActionsCacheList struct {
+	TotalCount    *int64          `json:"total_count,omitempty"`
+	ActionsCaches []*ActionsCache `json:"actions_caches,omitempty"`
+}
+
+// ActionsCacheUsage represents a GitHub Actions Cache Usage object.
+//
+//GitHub API docs: https://docs.github.com/en/rest/actions/cache?apiVersion=2022-11-28#get-github-actions-cache-usage-for-a-repository
+type ActionsCacheUsage struct {
+	FullName             *string `json:"full_name"`
+	ActiveCacheUsageSize *int64  `json:"active_caches_size_in_bytes"`
+	ActiveCachesCount    *int64  `json:"active_caches_count"`
+}
+
+// ActionsCacheUsageList represents a list repositories with GitHub Actions cache usage for an organization.
+//
+//GitHub API docs: https://docs.github.com/en/rest/actions/cache?apiVersion=2022-11-28#get-github-actions-cache-usage-for-a-repository
+type ActionsCacheUsageList struct {
+	TotalCount     *int64               `json:"total_count"`
+	RepoCacheUsage []*ActionsCacheUsage `json:"repository_cache_usages"`
+}
+
+// TotalCacheUsage represents total github actions cache usage of an organization or enterprise.
+//
+type TotalCacheUsage struct {
+	TotalActiveCacheUsageSize *int64 `json:"total_active_caches_size_in_bytes"`
+	TotalActiveCachesCount    *int64 `json:"total_active_caches_count"`
+}
+
+// ActionsCacheListOptions list all possible optional Query parameters for ListCaches method.
+// GitHub API docs:  https://docs.github.com/en/rest/actions/cache?apiVersion=2022-11-28#list-github-actions-caches-for-a-repository
+type ActionsCacheListOptions struct {
+	//The number of results per page (max 100). Default:30
+	PerPage int `url:"per_page,omitempty"`
+	//Page number of the results to fetch.Default:1
+	Page int `url:"page,omitempty"`
+	//The Git reference for the results you want to list.
+	// The ref for a branch can be formatted either as refs/heads/<branch name>
+	//or simply <branch name>. To reference a pull request use refs/pull/<number>/merge
+	Ref string `url:"ref,omitempty"`
+	Key string `url:"key,omitempty"`
+	//Can be one of: "created_at", "last_accessed_at", "size_in_bytes". Default: "last_accessed_at"
+	Sort string `url:"sort,omitempty"`
+	//Can be one of: "asc", "desc" Default: desc
+	Direction string `url:"direction,omitempty"`
```

Comment:
For this section, and all comments throughout, please add a single space after the `//`.